openbb_platform/extensions/platform_api/openbb_platform_api/response_models.py

```diff
@@ -147,3 +147,149 @@ def validate_model(cls, values) -> "PdfResponseModel":
         values.data_format = {"data_type": "pdf", "filename": filename}
 
         return values
+
+
+class OmniWidgetResponseModel(Data):
+    """Omni Widget Response Model.
+
+    Supply the content, and optionally the `parse_as` field.
+
+    Fields
+    ------
+    content : Any
+        The content to display in the Omni widget.
+    data_type : Optional[str]
+        The type of data, which is always "object" for Omni widgets, no need to specify.
+    parse_as : Optional[str]
+        The type of content to parse as. One of "table", "chart", or "text".
+        Attempts to set this automatically based on the content type, but can be overridden.
+
+    Returns
+    -------
+    object
+        Object that conforms to the validated output requirements of the API.
+
+    Example
+    -------
+    >>> from openbb_platform_api.main import app
+    >>> @app.get("/omni_widget", response_model=OmniWidgetResponseModel)
+    >>> async def get_omni_widget():
+    >>>     return {"content": [{"name": "Alice", "age": 30}, {"name": "Bob", "age": 25}]}
+    """
+
+    model_config = ConfigDict(
+        extra="ignore",
+        json_schema_extra={
+            "x-widget_config": {
+                "$.type": "omni",
+            }
+        },
+    )
+
+    content: Any = Field(
+        description="The content to display in the Omni widget.",
+        json_schema_extra={"x-widget_config": {"exclude": True}},
+    )
+    parse_as: Optional[str] = Field(
+        default=None,
+        description="The type of content to parse as. One of 'table', 'chart', or 'text'.",
+        json_schema_extra={"x-widget_config": {"exclude": True}},
+    )
+    data_format: Optional[dict] = Field(
+        default=None,
+        description="Leave this field empty. This is populated by the model_validator.",
+        json_schema_extra={"x-widget_config": {"exclude": True}},
+    )
+
+    @model_validator(mode="after")
+    @classmethod
+    def validate_model(  # pylint: disable=import-outside-toplevel
+        cls, values
+    ) -> "OmniWidgetResponseModel":
+        """Validate the Omni widget content."""
+        import contextlib  # noqa
+        import json
+        import pandas as pd
+
+        content = getattr(values, "content", None)
+
+        if content is None:
+            raise ValueError("Content cannot be empty.")
+
+        parse_as = getattr(values, "parse_as", None)
+
+        if parse_as and parse_as not in ("table", "chart", "text"):
+            raise ValueError(
+                "Invalid parse_as value. Must be one of 'table', 'chart', or 'text'."
+            )
+
+        # If parameter was supplied, assume the data is formatted correctly.
+        if content and parse_as:
+            data_format = {
```

Comment:
I couldn't follow this part too. 

Can you explain the logic of the validator in plain English?

Reply:
This allows the user to override the format discovery and assignment. If you supply a "parse_as" key, then it will assume you explicitly want that setting and will pass directly as-is.

Reply:
The shape of the input is not the same as the shape of the output, this guarantees that the output is correctly formatted for what Workspace expects.

Reply:
Resolving these conversations following the call we had.